### PR TITLE
xz 5.4.6

### PR DESCRIPTION
**Pull request by skupr-anaconda**
xz 5.4.6

**Destination channel:** defaults

### Links

- [PKG-4149](https://anaconda.atlassian.net/browse/PKG-4149) 
- [Upstream repository](https://github.com/tukaani-project/xz/tree/v5.4.6)
- [Upstream changelog/diff](https://github.com/tukaani-project/xz/compare/v5.4.5...v5.4.6)

### Explanation of changes:

- Copy the pkgconfig file `liblzma.pc` on `win`
- Update `source/url` because `WARNING: Error: HTTP 404 NOT FOUND for url <https://sourceforge.net/projects/lzmautils/files/xz-5.6.0.tar.bz2>`
- Add new test commands
- Update home url
- Update dev_url

### Notes

- It was found that `liblzma.pc` file is missing on **win64** but it's required by `r-cairo` package. `r-cairo` has a couple of problems:
  - it really wants a `PKG_CONFIG_PATH` to point to `%LIBRARY_LIB%/pkgconfig` to pick up `cairo`, `libtiff` and `libjpeg` info
  - `libtiff` wants to use `liblzma` (which _MSYS2_ supplies separately as `msys2-liblzma` but is bundled in `xz` in **defaults**) and the defaults package doesn't have a `liblzma.pc `file -- it is created during a build but not obviously copied for install.
- We do not build the latest version **5.6.0** yet because of the significant changes, see https://github.com/tukaani-project/xz/releases/tag/v5.6.0.

[PKG-4149]: https://anaconda.atlassian.net/browse/PKG-4149?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by cbouss** [comment]
It doesn't have to be this build , but we can build for 5.6
```
5.6.0 (2024-02-24)

    This bumps the minor version of liblzma because new features were
    added. The API and ABI are still backward compatible with liblzma
    5.4.x and 5.2.x and 5.0.x.

    NOTE: As described in the NEWS for 5.5.2beta, the core components
    are now under the BSD Zero Clause License (0BSD).
```
And our pinning is on the major, so doesn't require a rebuild.

**Review by cbouss — COMMENTED** [review]
The windows pc files content is as such:

```
prefix=/c/b/abs_354ppdp63_/croot/xz_1708953363008/_h_env
exec_prefix=/c/b/abs_354ppdp63_/croot/xz_1708953363008/_h_env
libdir=/c/b/abs_354ppdp63_/croot/xz_1708953363008/_h_env/lib
includedir=/c/b/abs_354ppdp63_/croot/xz_1708953363008/_h_env/include
```

As far as I remember, while the prefix is replaced at runtime, for it to be utilize correctly, libdir and includedir should reference the prefix. (Something like `libdir=${prefix}/lib`)

**Comment by skupr-anaconda** [comment]
@cbouss I've updated the version to **5.6.0** and added new test commands for `pkg-config`

**Comment by skupr-anaconda** [comment]
@cbouss v5.6.0 fails on windows because the maintainers removed build files https://github.com/tukaani-project/xz/compare/v5.4.5...v5.6.0#diff-a8a8e71ac794800214e0e9b83eb1d64f849720f090c61e5d3413e1c75d2f8ec1 and now we need another way to build it.
As the MSYS2/R stuff can be blocked by this obstacle I would suggest building v5.4.6 and opening a JIRA ticket for v5.6.0+

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by SC426 — APPROVED** [review]

**Review by cbouss — APPROVED** [review]
LGTM

I wasn't sure about the pc file on windows. After checking I get this:
```
(testxz) PS C:\Windows\system32> pkg-config --variable=libdir --dont-define-prefix liblzma
/c/b/abs_45ot_z2bh8/croot/xz_1708965552287/_h_env/lib
(testxz) PS C:\Windows\system32> pkg-config --variable=prefix liblzma
C:/miniconda3/envs/testxz/Library
(testxz) PS C:\Windows\system32> pkg-config --variable=exec_prefix liblzma
/c/b/abs_45ot_z2bh8/croot/xz_1708965552287/_h_env
(testxz) PS C:\Windows\system32> pkg-config --variable=libdir liblzma
C:/miniconda3/envs/testxz/Library/lib
(testxz) PS C:\Windows\system32> pkg-config --variable=includedir liblzma
C:/miniconda3/envs/testxz/Library/include
```

To note: `exec_prefix` will not be usable, but as far as I can see, there is not impact for this package. The package won't work if we set `--dont-define-prefix` (which should not be set by default anyway).

**Pull request merged**
